compiler/src/dotty/tools/dotc/cc/CheckCaptures.scala

```diff
@@ -168,6 +168,13 @@ object CheckCaptures:
             if !seen.contains(t) then
               seen += t
               traverseChildren(t)
+
+              // Check the lower bound of path dependent types.
+              // See issue #19330.
+              val isTypeParam = t.prefix eq NoPrefix
```

Comment:
`isTypeParam` is misleading as a name. It could also be a local type definition, that also will lead to a `NoPrefix`. I'd turn it around and define:
```scala
val isMember = t.prefix ne NoPrefix
```